src/DSi_DSP.cpp

```diff
@@ -0,0 +1,591 @@
+/*
+    Copyright 2020 PoroCYon
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#include "teakra/include/teakra/teakra.h"
+
+#include "DSi.h"
+#include "DSi_DSP.h"
+#include "FIFO.h"
+#include "NDS.h"
+
+static constexpr u32 DataMemoryOffset = 0x20000; // from Teakra memory_interface.h
+// NOTE: ^ IS IN DSP WORDS, NOT IN BYTES!
+
+namespace DSi_DSP
+{
+
+Teakra::Teakra* TeakraCore;
+
+bool SCFG_RST;
+
+u16 DSP_PADR;
+u16 DSP_PCFG;
+u16 DSP_PSTS;
+u16 DSP_PSEM;
+u16 DSP_PMASK;
+u16 DSP_PCLEAR;
+u16 DSP_CMD[3];
+u16 DSP_REP[3];
+
+u64 DSPTimestamp;
+
+FIFO<u16, 16> PDATAReadFifo/*, *PDATAWriteFifo*/;
+int PDataDMALen = 0;
+
+static u16 GetPSTS()
```

Comment:
why is the static here?

Reply:
remind me what static does again? i dont think it needs to be static that was ust a side eect o some tracking code. i can revert it to non-static.

Reply:
i think that might o been a change in the codebase a year ago, should i remove static rom all methods?

Reply:
static shouldn't be necessary because we use C++ namespaces